docs/presets/icons.md

```diff
@@ -298,6 +294,101 @@ presetIcons({
 })
 ```
 
+## Options
+
+### scale
+
+- Type: `number`
+- Default: `1`
+
+Scale related to the current font size (1em).
+
+### mode
+
+- Type: `'mask' | 'background-img' | 'auto'`
+- Default: `'auto'`
+- See: https://antfu.me/posts/icons-in-pure-css
+
+Mode of generated CSS icons.
+
+:::tip
+- `mask` - use background color and the `mask` property for monochrome icons
+- `background-img` - use background image for the icons, colors are static
+- `auto` - smartly decide mode between `mask` and `background-img` per icon based on its style
+:::
+
+### prefix
+
+- Type: `string | string[]`
+- Default: `'i-'`
+
+Class prefix for matching icon rules.
+
+### extraProperties
+
+- Type: `Record<string, string>`
+- Default: `{}`
+
+Extra CSS properties applied to the generated CSS.
+
+### warn
+
+- Type: `boolean`
+- Default: `false`
+
+Emit warning when missing icons are matched.
+
+### collections
+
+- Type: `Record<string, (() => Awaitable<IconifyJSON>) | undefined | CustomIconLoader | InlineCollection>`
+- Default: `undefined`
+
+In Node.js environment, the preset will search for the installed iconify dataset automatically. When using in the browser, this options is provided to provide dataset with custom loading mechanism.
+
+### layer
+
+- Type: `string`
+- Default: `'icons'`
+
+Rule layer.
+
+### customizations
+
+- Type: `Omit<IconCustomizations, 'additionalProps' | 'trimCustomSvg'>`
+- Default: `undefined`
+
+Custom icon customizations.
+
+### autoInstall
+
+- Type: `boolean`
+- Default: `false`
+
+Auto install icon sources package when the usages is detected.
+
+:::warning
+Only on `node` environment, on `browser` this option will be ignored.
+:::
+
+### unit
+
+- Type: `string`
+- Default: `'em'`
+
+Custom icon unit.
+
+### cdn
```

Comment:
I need to review current state for runtime, maybe we need to add some more hints here

Reply:
the PR LGTM, can be added later if necessary

Reply:
https://github.com/unocss/unocss/pull/1555